atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyUnicastService.java

```diff
@@ -94,18 +101,20 @@ public void unicast(final Address address, final String subject, final byte[] pa
       LOGGER.debug("Failed sending unicast message, unicast service was not started.");
       return;
     }
-    final InetAddress resolvedAddress = address.tryResolveAddress();
-    if (resolvedAddress == null) {
-      LOGGER.debug(
-          "Failed sending unicast message (destination address {} cannot be resolved)", address);
-      return;
-    }
+    final Future<InetSocketAddress> inetSocketAddress =
```

Comment:
I researched this for a good while but came to the conclusion that this (meaning sending UDP packets to unresolved addresses when a dns resolver is part of the pipeline) is simply not supported by Netty :disappointed:

Reply:
I think the Netty way of doing this is to implement DNS resolution as part of your outbound pipeline, actually.

Reply:
Using ResolveAddressHandler for example

Reply:
We can leave it as is, since the handler does the asynchronous thing I suggested above. It would still be nice to have it async though, but it's still only a suggestion :+1: